### PR TITLE
PLAT-7275 Fallback fixes

### PR DESCRIPTION
**Pull request by miguelhar**
[PLAT-7275](https://dominodatalab.atlassian.net/browse/PLAT-7275)
* Sometimes the executor does not have perms to chmod the private key(`/root/pvt-key.pem`).
* Adds a retry on open ssh tunnel.

### CODE REVIEW

**Review by fraenkel — COMMENTED** [review]
Even if it lives in root, you are saying we aren't executing as root and therefore the permissions may also be incorrect and fail on the next line.

**Comment by miguelhar** [comment]
> Even if it lives in root, you are saying we aren't executing as root and therefore the permissions may also be incorrect and fail on the next line.

Next line is a read which can/should work, what am i missing?

**Comment by fraenkel** [comment]
ssh will fail if the permissions are too wide.

**Comment by miguelhar** [comment]
> ssh will fail if the permissions are too wide. otherwise why are we even `chmod`ing.

thats is true but given that the permissions fall in the middle it will work.(440)

```
#cant change perms
$ chmod 400 test.pem
chmod: Unable to change file mode on test.pem: Operation not permitted

# but can ssh
 ssh -oIdentitiesOnly=yes -i test.pem -oUserKnownHostsFile=/dev/null -oStrictHostKeyChecking=no ec2-user@34.211.197.67 -- hostname
Warning: Permanently added '34.211.197.67' (ED25519) to the list of known hosts.

ip-10-0-0-29.us-west-2.compute.internal
```

We chmod to be overly cautious when we are the ones that create the key

File: modules/eks/submodules/k8s/templates/k8s-pre-setup.sh.tftpl [hunk 1]
```diff
@@ -1,12 +1,29 @@
...
+      echo "Attempt $i failed. Retrying in $sleep_time seconds..."
+      sleep $sleep_time
+    fi
+  done
+
+  if [[ $success -eq 0 ]]; then
+    echo "Failed to open SSH tunnel after $max_retries attempts. Exiting..."
+    exit 1
+  fi
+}
```
Comment (fraenkel) [inline modules/eks/submodules/k8s/templates/k8s-pre-setup.sh.tftpl:22]: seems a bit overly complicated
```
for i in $(seq 1 $max_retries); do
    open_ssh_tunnel_to_k8s_api && return 0
   [ "$i" = "$max_retries" ] && echo "Failed to open ....." && return 1
  echo "Attempt $i failed. Retrying in $sleep_time seconds..."
  sleep "$sleep_time"
done
```

**Review by fraenkel — APPROVED** [review]

**Pull request merged**